### PR TITLE
Adding disclaimer about consistency (fencing tokens and ttls expirati…

### PR DESCRIPTION
**Pull request by jhmilan**
…on mechanism

### CODE REVIEW

**Comment by jhmilan** [comment]
@oranagra I created this PR as discussed here https://github.com/redis/redis/pull/7644#issuecomment-1206815232

**Review by oranagra — COMMENTED** [review]
LGTM.. not sure if this part is needed:

> and there are no plans to use them
for now (August, 2022)](https://github.com/redis/redis/pull/7644#issuecomment-1206815232)

File: docs/reference/patterns/distributed-locks.md [hunk 1]
```diff
@@ -206,6 +206,23 @@ However this does not technically change the algorithm, so the maximum number
...
+quite some time and it is not a problem only happening in Redlock but in any distributed locks system.
+Extending locks can be also be an option, but please don´t assume that a lock will be retained forever
+when the process which acquired it is still alive.
+2. [Redis is not using monotonic clock for TTL expiration mechanism and there are no plans to use them
+for now (August, 2022)](https://github.com/redis/redis/pull/7644#issuecomment-1206815232). There might be
+one scenario where a lock can be acquired by more than 1 process after some wall-clock shift. Even though
+the problem can be mitigated or even avoided by a) not allowing admins to change date manually, and also
+b) properly setting up NTP, it is still possible that this issue would happen in real life and consistency could be 
+compromised.
+
```
Comment (itamarhaber) [inline docs/reference/patterns/distributed-locks.md:225]:
```suggestion
### Disclaimer about consistency

Please consider thoroughly reviewing the [Analysis of Redlock](#analysis-of-redlock) section at the end of this page.
Martin Kleppman's article and antirez's answer to it are very relevant.
If you are concerned about consistency and correctness, you should pay attention to the following topics:

1. You should implement fencing tokens.
  This is especially important for processes that can take significant time and applies to any distributed locking system.
  Extending locks' lifetime is also an option, but don´t assume that a lock is retained as long as the process that had acquired it is alive.
2. ["Redis is not using monotonic clock for TTL expiration mechanism and there are no plans to use them
for now" (August, 2022)](https://github.com/redis/redis/pull/7644#issuecomment-1206815232).
  That means that a wall-clock shift may result in a lock being acquired by more than one process.
  Even though the problem can be mitigated by preventing admins from manually setting the server's time and setting up NTP properly, there's still a chance of this issue occurring in real life and compromising consistency.

```

**Comment by itamarhaber** [comment]
Thanks for the contribution @jhmilan - I've suggested some edits. Also, please add the relevant words to the wordlist file so the spellchecker ignores these.

**Comment by jhmilan** [comment]
Thanks for the feedback @oranagra @itamarhaber 
I will update my PR as soon as I have 5 minutes!

**Comment by jhmilan** [comment]
@itamarhaber @oranagra PR is updated (Applied suggested changes and added words to wordlist for spellchecker)

File: docs/reference/patterns/distributed-locks.md [hunk 2]
```diff
@@ -206,6 +206,20 @@ However this does not technically change the algorithm, so the maximum number
...
+Please consider thoroughly reviewing the [Analysis of Redlock](#analysis-of-redlock) section at the end of this page.
+Martin Kleppman's article and antirez's answer to it are very relevant.
+If you are concerned about consistency and correctness, you should pay attention to the following topics:
+
+1. You should implement fencing tokens.
+  This is especially important for processes that can take significant time and applies to any distributed locking system.
+  Extending locks' lifetime is also an option, but don´t assume that a lock is retained as long as the process that had acquired it is alive.
+2. ["Redis is not using monotonic clock for TTL expiration mechanism and there are no plans to use them
+for now" (August, 2022)](https://github.com/redis/redis/pull/7644#issuecomment-1206815232).
```
Comment (oranagra) [inline docs/reference/patterns/distributed-locks.md:219]: @itamarhaber don't you think this part is excessive?
```suggestion
2. Redis is not using monotonic clock for TTL expiration mechanism.
```

Comment (jhmilan) [reply docs/reference/patterns/distributed-locks.md:219]: @itamarhaber @oranagra the reason for that "for now" and the link is, from may POV, a try to given some more context to a reader that might be wondering is the monotonic clock topic is something that eventually could happen. Basically the path I followed to that issue and this PR.
If you guys consider this too much I'm fine applying @oranagra suggestion.
I have very limited access to my laptop this week, though 😅

Comment (oranagra) [reply docs/reference/patterns/distributed-locks.md:219]: I think that generally the docs specify how things work, and not how they could have worked and why it's not that way, what was planned or decided against...

@itamarhaber can make that decision and apply / edit the text if needed.

Comment (itamarhaber) [reply docs/reference/patterns/distributed-locks.md:219]: I admired @jhmilan research capabilities in finding the discussion, but on 2nd thought I tend to agree with @oranagra - sometimes less is better.

**Review by itamarhaber — APPROVED** [review]

**Comment by jhmilan** [comment]
Glad to see this was finally merged.

Cheers guys!

**Pull request merged**